builtins: implement more PostGIS tutorial available builtins

otan opened the pull request:
Release note (sql change): Implemented the following builtins available
on Geometry types:
* ST_Area2D (resolves #48869)
* ST_NRings (resolves #48996)
* ST_PointN (resolves #49008)
* ST_GeometryType (resolves #48946)

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/49649)
<!-- Reviewable:end -->

sumeerbhola reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

> Reviewed 2 of 2 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/49649#-:-M8RWNx48d7RNgZPt2N9:bg7igkv)* status: :shipit: complete! 1 of 0 LGTMs obtained



<!-- Sent from Reviewable.io -->

otan committed:
builtins: implement more PostGIS tutorial available builtins

Release note (sql change): Implemented the following builtins available
on Geometry types:
* ST_Area2D (resolves #48869)
* ST_NRings (resolves #48996)
* ST_PointN (resolves #49008)
* ST_GeometryType (resolves #48946)

otan commented:
TFTR!

bors r=sumeerbhola
